app/components/goodcity/add-item-overlay.js

```diff
@@ -0,0 +1,85 @@
+import Ember from "ember";
+import AsyncMixin, { ERROR_STRATEGIES } from "stock/mixins/async";
+import _ from "lodash";
+
+export default Ember.Component.extend(AsyncMixin, {
+  packageService: Ember.inject.service(),
+  store: Ember.inject.service(),
+  messageBox: Ember.inject.service(),
+  i18n: Ember.inject.service(),
+
+  pkgLocations: Ember.computed("pkg.packagesLocations", function() {
+    return this.getAttributesFor("packagesLocations");
+  }),
+
+  pkgLocationName: Ember.computed("pkg.packagesLocations", function() {
+    return this.getAttributesFor("packagesLocationsName");
+  }),
+
+  getAttributesFor(parameter) {
+    let pkg = this.get("pkg");
+    if (pkg) {
+      if (parameter == "packagesLocationsName") {
+        return this.get("pkgLocations").get("firstObject.location.name");
+      }
+      return this.get("pkg").get("packagesLocations");
+    }
+  },
+
+  calculateSumFor(parameter) {
+    let quantities = [];
+    if (this.get("pkg.packagesLocations")) {
+      quantities = this.get("pkg.packagesLocations").map(value =>
+        value.get(parameter)
+      );
+    }
+
+    return quantities.reduce((accumulator, value) => {
+      return accumulator + (!(value == "") && parseInt(value));
+    }, 0);
+  },
+
+  totalNumberTomove: Ember.computed(
+    "pkg.packagesLocations.@each.defaultQuantity",
+    function() {
+      return this.calculateSumFor("defaultQuantity");
+    }
+  ),
+
+  addRemoveItem(params) {
+    this.runTask(async () => {
+      const data = await this.get("packageService").addRemoveItem(
+        this.get("entity.id"),
+        params
+      );
+      this.get("store").pushPayload(data);
+      this.set("open", false);
+      this.sendAction("onConfirm");
+    }, ERROR_STRATEGIES.MODAL);
+  },
+
+  actions: {
+    moveItemToBox() {
+      let pkg = this.get("pkg");
+      if (pkg) {
+        this.get("pkgLocations").forEach(pkgLocation => {
+          let selectedQuantity = pkgLocation.get("defaultQuantity");
+          if (pkgLocation.get("hasDirtyAttributes") && selectedQuantity !== 0) {
```

Comment:
```suggestion
          if (pkgLocation.get("hasDirtyAttributes") && selectedQuantity) {
```

Reply:
@swatijadhav this is intentionally done, i want to proceed only if selectedQuantity is not 0. `selectedQuantity !== 0`.

Reply:
ohh, understood, `if(true && 0)` will go to else. 
Thanks @swatijadhav